ompi/mca/common/ompio/common_ompio_file_read.c

```diff
@@ -132,7 +131,7 @@ int mca_common_ompio_file_read (ompio_file_t *fh,
     else {
 	bytes_per_cycle = OMPIO_MCA_GET(fh, cycle_buffer_size);
     }
-    cycles = ceil((float)max_data/bytes_per_cycle);
+    cycles = (max_data + bytes_per_cycle - 1) / bytes_per_cycle;
```

Comment:
As @ax3l wrote [here](https://github.com/open-mpi/ompi/pull/6286#discussion_r248764855) I would suggest to use `uint64_t` at this place else this equation will reduce the possible range of `max_data` on a 32Bit system to `[0;max_data - bytes_per_cycle)`. All values larger for `max_data` will produce an overflow. 
With the current PR it would not be possible to write ~4GiB data on a 32Bit system.

Reply:
A possible other solution if the type should stay `size_t` is: `cycles = max_data / bytes_per_cycle + max_data % bytes_per_cycle ? 1u : 0u ;`

Reply:
Well, honestly I think going back to the ceil solution might the `easiest way out, and simple avoid the type cast to a float. The problem is that changing the 'cycles' variable to uint64_t forces you to change that also for the 'index; variable (otherwise you get a warning in the for loop), and the interface of the build_io_array function which is used all over the code base. Not sure I want to make this sweeping change at this point.

Reply:
Another solution could be to calculate the no of cycles into a temporary a uint64_t, and then assign that value using an int typcast to 'cycles'. This would have the advantage that we do not have to change the interfaces, and till get rid of the ceil function. An int as the datatype for the no_of_cycles is perfectly fine, you can read/write 512 MB per cycle, so (if my math is correct) that would still allow for 1024 PB of data overall being written, that should be good enough for the next decade.

Reply:
Rather the latter, if you ask me, but I can check tomorrow. Be aware that there is no int-ceil in C, which would mean even without the cast it is an implicit cast to a float type...

What's the exact warning you see now?

Reply:
@edgargabriel have you seen this comment? :) We just mentioned that `max_data` on 32-bit systems will limit writes to 4 Gbyte as it is only `size_t`. That's our concern.

Regarding the `cycles` var: it should probably also be at least of `size_t`, better also a fixes large integer type as you just added. An `int` is most of the time 2 to 4 bytes. With an upper bound of 32767 cycles on some 32-bit systems and 512 MByte per cycle this is as well "only" 16 PByte. (4 Byte `int`s are indeed opening this to 1 EByte). Not as critical here, but `max_data` is a concern and should be of larger type.